src/delphyne/model/custom_vocab/base_manager/concept_manager.py

```
@@ -72,15 +98,27 @@ def _load_custom_concepts(self, vocab_ids: Set[str], valid_prefixes: Set[str]) -
                     concept_id = row['concept_id']
                     vocabulary_id = row['vocabulary_id']
 
+                    # if file prefix is valid vocab_id,
+                    # vocabulary_ids in file should match it.
+                    # comparison is case-insensitive.
+                    if check_prefix and vocabulary_id.lower() != file_prefix:
+                        invalid_vocabs.add(vocabulary_id)
+
                     # quality checks
+                    # NOTE: only performing checks not already enforced
+                    # by CONCEPT table constraints; those will raise an
+                    # exception as soon as they occur
                     if int(concept_id) < 2000000000:
-                        raise ValueError(
-                            f'{concept_file.name} must have concept_ids starting at '
-                            f'2\'000\'000\'000 (2B+ convention)')
+                        errors.add(f'{concept_file.name} must contain concept_ids starting at '
+                                   f'2\'000\'000\'000 (2B+ convention)')
+                        file_errors = True
                     if concept_id in unique_concepts_check:
-                        raise ValueError(
-                            f'concept {concept_id} has duplicates across one or multiple '
-                            f'files')
+                        errors.add(f'concept {concept_id} is duplicated across one or multiple '
+                                   f'files')
+                        file_errors = True
+                        continue
+
+                    unique_concepts_check.add(concept_id)
 
                     if vocabulary_id in vocab_ids:
```

Could be useful to also log a warning when an unknown vocabulary_id (not present in any vocabulary.tsv file) is encountered in a concept file. Now they are silently ignored.

---

Added warning and updated tests